### ISSUE TITLE
regression in 2.0.3 in deno compile from stdin

### ISSUE THREAD

**Issue by Ambyjkl** [denoland/deno#26593]
Version: Deno 2.0.3
Last known good version: Deno 2.0.2

Run the following:
```
deno compile --ext js -o helloworld /dev/stdin <<<'console.log("hello world")'
```
Actual: throws a file not found error
```
Compile file:///dev/stdin to helloworld
error: Writing deno compile executable to temporary file 'helloworld.tmp-7ca31622898c7656'

Caused by:
    0: Failed adding '/dev/stdin'
    1: No such file or directory (os error 2)
```
Expected: succeeds with no error (in Deno 2.0.2)
```
Compile file:///dev/stdin to helloworld
```

**Comment by Albinzr** [comment]
Its mensioned 2.1 is stable ... and I tried this also still having same error. I think Its better to stay with node.js. I'm seeing all this promotion for deno 2 but there is no stability at all.

**Comment by dsherret** [comment]
@Albinzr surprising the issue is not fixed when the issue is still open.

**Comment by dsherret** [comment]
I'm actually going to close this issue as won't fix. This scenario was never documented and just happened to work before, but it's too much to support.

**Comment by Albinzr** [comment]
@dsherret I'm a product person, and from a product standpoint, this is a significant issue: Deno, including the 2.x version, is unable to compile. When companies want to test things out and discover that basic features, which were prominently highlighted in all the promotions for Deno 2.x, are not functioning as expected, it will lead to a direct drop in interest.

**Comment by Ambyjkl** [comment]
> I'm a product person

@Albinzr I'm going to assume you're semi/non-technical and don't know what stdin is. Deno 2.x is still able to compile in 99% of real use cases, this is just a fringe use case that caught my eye. This is not a blocker for adopting Deno 2.x, you are free to switch over from Node.js if you wish, or just 

For more context, I was testing binary sizes of `deno compile` across different versions, and I was passing in the input over stdin because I was too lazy to create a new file, and that is how I stumbled across this.

**Comment by dsherret** [comment]
Yeah, this scenario seems very fringe to me and it’s easy to work around with a temp file. Still 3 steps instead of 7 https://nodejs.org/api/single-executable-applications.html#single-executable-applications

Also, this was not highlighted anywhere and isn’t a significant issue. If there are any use cases that are blocked by this I’d be open to reopening, but I feel like it’s not worth supporting because of the added complexity. There’s a lot more stuff that works in the latest `deno compile` that doesn’t work anywhere else. Also, good to hear you’re not blocked on this one, @Ambyjkl.
